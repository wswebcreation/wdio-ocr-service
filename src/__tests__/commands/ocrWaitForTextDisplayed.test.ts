import ocrWaitForTextDisplayed from '../../commands/ocrWaitForTextDisplayed'
import ocrGetText from '../../commands/ocrGetText'

jest.mock('../../commands/ocrGetText', ()=> jest.fn().mockResolvedValue(''))

describe('ocrWaitForTextDisplayed', () => {
  it('should be able to wait for a text to be displayed', async () => {
    const options = {
      isTesseractAvailable: true,
      ocrImagesPath: 'ocrImagesPath',
      language: 'eng',
      reuseOcr: true,
      screenSize: { width: 1, height: 2 },
      text: 'text',
    }
    const globalAny:any = global
    globalAny.driver = {
      waitUntil: jest.fn().mockImplementation((fn)=>fn()),
    }

    await ocrWaitForTextDisplayed(options)

    expect(await globalAny.driver.waitUntil).toHaveBeenCalledTimes(1)
    expect(await ocrGetText).toHaveBeenCalledWithSnapshot()
  })
})
