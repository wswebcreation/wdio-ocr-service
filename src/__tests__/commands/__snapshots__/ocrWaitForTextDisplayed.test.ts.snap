// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ocrWaitForTextDisplayed should be able to wait for a text to be displayed 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "androidRectangles": undefined,
        "iOSRectangles": undefined,
        "isTesseractAvailable": true,
        "language": "eng",
        "ocrImagesPath": "ocrImagesPath",
        "reuseOcr": false,
        "screenSize": Object {
          "height": 2,
          "width": 1,
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;
